src/firmware/shared/math/tbots_math.h

```diff
@@ -7,3 +7,26 @@
 
 // define our own PI value here that is a float because M_PI in math.h is a double
 #define P_PI 3.14159265f
+
+/**
+ * Function performs a linear interpolation between the points [x0 ,y0] and [x1, y1] at
+ * the position xp
+ *
+ * @pre x1 < xp < x2, or in other words - xp must be between x0 and x1 AND x0 and x1
```

Comment:
if xp is out of the range `[x1, x2]` you could always clamp to the min or max y value. Not a blocking comment just a suggestion.

Reply:
Added clamping and tests 👍